cluster/consensus_module_proxy_compat.go

```diff
@@ -0,0 +1,182 @@
+package cluster
+
+import (
+	"github.com/lirm/aeron-go/aeron/atomic"
+	"github.com/lirm/aeron-go/aeron/logbuffer"
+	"github.com/lirm/aeron-go/cluster/codecs"
+)
+
+// -----------------------------------------------------------------------------
+// Java compat implementation and using TryClaim for better performance
+// https://github.com/real-logic/aeron/blob/release/1.46.x/aeron-cluster/src/main/java/io/aeron/cluster/service/ConsensusModuleProxy.java
+// -----------------------------------------------------------------------------
+
+// https://github.com/real-logic/aeron/blob/release/1.46.x/aeron-cluster/src/main/java/io/aeron/cluster/service/ConsensusModuleProxy.java#L146
+func (proxy *consensusModuleProxy) ack(
+	logPosition int64,
+	timestamp int64,
+	ackID int64,
+	relevantID int64,
+	serviceID int32,
+) (bool, error) {
+	length := int32(new(codecs.MessageHeader).EncodedLength() + int64(new(codecs.ServiceAck).SbeBlockLength()))
+	position := proxy.publication.TryClaim(length, &proxy.bufferClaim)
+	if position > 0 {
+		// Create a packet and send it
+		bytes, err := codecs.ServiceAckRequestPacket(
+			proxy.marshaller,
+			proxy.rangeChecking,
+			logPosition,
+			timestamp,
+			ackID,
+			relevantID,
+			serviceID,
+		)
+		if err != nil {
+			proxy.bufferClaim.Abort()
+			return false, err
+		}
+
+		proxy.bufferClaim.Buffer().PutBytesArray(proxy.bufferClaim.Offset(), &bytes, 0, length)
+		proxy.bufferClaim.Commit()
+		return true, nil
+	}
+
+	if err := checkResult(position, proxy.publication); err != nil {
+		proxy.bufferClaim.Abort()
+		return false, err
+	}
+	return false, nil
+}
+
+func (proxy *consensusModuleProxy) closeSession(
+	clusterSessionId int64,
+) (bool, error) {
+	length := int32(new(codecs.MessageHeader).EncodedLength() + int64(new(codecs.CloseSession).SbeBlockLength()))
+	position := proxy.publication.TryClaim(length, &proxy.bufferClaim)
+	if position > 0 {
+		bytes, err := codecs.CloseSessionRequestPacket(
+			proxy.marshaller,
+			proxy.rangeChecking,
+			clusterSessionId,
+		)
+		if err != nil {
+			proxy.bufferClaim.Abort()
```

Comment:
where is this called in java?

Reply:
Just a clean up mechanism in go vs autocloseable in java